### PR TITLE
BREAKING CHANGE: add support to log tenantId

### PR DESCRIPTION
**Pull request by Substancia**
Includes handling of missing contexts.

Evidences:-
MT with context: [https://app.flonnect.com/view/video/jibin.m/63919070f31aaa756ef77d3d](https://app.flonnect.com/view/video/jibin.m/63919070f31aaa756ef77d3d)
MT without context: [https://app.flonnect.com/view/video/jibin.m/63919100f31aaa756ef77d5f](https://app.flonnect.com/view/video/jibin.m/63919100f31aaa756ef77d5f)
ST: [https://app.flonnect.com/view/video/jibin.m/6391944cf31aaa756ef77df4](https://app.flonnect.com/view/video/jibin.m/6391944cf31aaa756ef77df4)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "logger-safe-security",
-  "version": "1.2.1",
+  "version": "1.2.2",
```
Comment (amanjain14) [inline package.json]: Revert this change. We have auto-release configured.

File: src/index.ts [hunk 2]
```diff
@@ -18,16 +27,44 @@ const formatError = winston.format(info => {
...
+            const { valueFromMethod } = parameters.tenantId;
+            try {
+                if (
+                    typeof valueFromMethod === "function" &&
+                    typeof valueFromMethod<string>("tenantId") === "string"
+                ) {
+                    info.tenantId = valueFromMethod<string>("tenantId");
+                }
+            } catch (error) {
+                console.error(
```
Comment (amanjain14) [inline src/index.ts]: Remove the error log.

File: src/index.ts [hunk 3]
```diff
@@ -18,16 +27,44 @@ const formatError = winston.format(info => {
     return info;
 });
 
+/* A custom format that is used to include tenantId. */
+const formatTenantId = (parameters: ParameterConfig | undefined) => {
+    return winston.format(info => {
+        info.tenantId = "";
```
Comment (amanjain14) [inline src/index.ts]: Remove this as well. If we get tenantId from the context then use it, else make no changes.

File: src/index.ts [hunk 4]
```diff
@@ -18,16 +27,44 @@ const formatError = winston.format(info => {
     return info;
 });
 
+/* A custom format that is used to include tenantId. */
+const formatTenantId = (parameters: ParameterConfig | undefined) => {
+    return winston.format(info => {
```
Comment (amanjain14) [inline src/index.ts:40]: Can we remove the dependency on the tenantId and make this generic?
From the parameters config, get the keys, check if the key is present in the store, if yes, add it to `info` else skip.

**Review by amanjain14 — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 4 above)
Comment (amanjain14) [reply src/index.ts:40]: Please update the PR title with breaking changes.

**Comment by Substancia** [comment]
To test this:-

New `createLogger` method call in `<container>/src/config/logger.ts` is
```
const logger = createLogger({
    logLevel: APP_LOG_LEVEL,
    service: process.env.SERVICE_NAME,
    config: {
        parameters: {
            tenantId: {
                valueFromMethod: getValueFromStore
            }
        }
    }
});
```

**Review by aneeshrelan — APPROVED** [review]
LGTM

**Pull request merged**